http-server-netty/src/main/java/io/micronaut/http/server/netty/binders/NettyBodyAnnotationBinder.java

```diff
@@ -159,4 +163,38 @@ Optional<T> transform(NettyHttpRequest<?> nhr, ArgumentConversionContext<T> cont
             .convert(conversionService, context)
             .map(o -> (T) o.claimForExternal());
     }
+
+    @Override
+    public Optional<Prepared> prepare(Argument<T> argument, MatchRule.ContentType fixedContentType) {
```

Comment:
since this is going to be an internal non-public API do we need the overhead of `Optional`? Can we just make the return value nullable and return null?

Reply:
either is fine but this is not a hot path. this is called once per route, so i think it's better to have a nicer api here than to overoptimize this.